src/prism-async.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import highlight from './highlight';
+
+const languages = [];
+
+export const registerLanguage = (_, language) => {
+  if(ReactHighlightAsync.refractor) {
+    return refractor.register(language);
+  } else {
+    languages.push(language);
+  }
+};
+
+export default class ReactHighlightAsync extends React.PureComponent {
+  static refractor = null;
+  static highlightInstance = (highlight(null, {}));
+  static refractorPromise = null;
+  
+  static _loadRefractor() {
+    ReactHighlightAsync.refractorPromise = import(/* webpackChunkName:"react-syntax-highlighter/refractor-import" */ 'refractor/core').then(({ default: refractor }) => {
+      languages.forEach((language) => refractor.register(language));
+      ReactHighlightAsync.refractor = refractor;
+    });
+    
+    return ReactHighlightAsync.refractorPromise;
+  }
+
+  componentWillMount() {
```

Comment:
can we use didMount since this will eventually go away -- https://reactjs.org/docs/react-component.html#legacy-lifecycle-methods

Reply:
Done